src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs

```diff
@@ -42,14 +43,16 @@ public void InvalidateCacheDependencies(EFCacheKey cacheKey)
 
 public class EFCacheKeyMarshaller : Marshaller<EFCacheKey>
 {
+    // DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(EFCacheKey));
     public override byte[] marshall(EFCacheKey t)
     {
-        throw new NotImplementedException();
+        return JsonSerializer.SerializeToUtf8Bytes(t);
     }
 
     public override EFCacheKey unmarshall(byte[] buff)
     {
-        throw new NotImplementedException();
+        var utf8Reader = new Utf8JsonReader(buff);
+        return JsonSerializer.Deserialize<EFCacheKey>(ref utf8Reader);
```

Comment:
do we really need the `ref` keyword? I don't think `Deserialize` wants to change the reader itself.

Reply:
I think we need to use `ref` keyword. Its mentioned in the official documentation and I also tried removing the keyword but got error.

Reply:
oh I see, good then